src/app/services/progressesApi.ts

```diff
@@ -1,5 +1,10 @@
 import { api } from './api';
-import { userDetails } from '@/types/standup.type';
+import { progressDetails } from '@/types/standup.type';
+
+type queryParamsType = {
```

Comment:
If I understand correctly, in one request either taskId will be a param or userId, not both.
The current type does not enforce that.
Below type will enforce it
```
type queryParamsType = {
    userId: string;
} |  {
    taskId: string;
} 
```
If both params can be passed together then ignore this comment.

Reply:
I am not sure how would I handle it in the place where I am using it below.

Reply:
because with this i will have to make 2 RTK Queries making me write the repetitive code again

Reply:
I guess if you change your code like below it should work

```
query:  (param: queryParamsType) => {
return {
                 url: '/progresses',
                 params: param,
           };
}
```

Still if you this helps, else no issues with the current code.